AndroidAnnotations/androidannotations-core/androidannotations-testutils/src/main/java/org/androidannotations/testutils/AAProcessorTestHelper.java

```diff
@@ -39,7 +41,20 @@ public File toGeneratedFile(Class<?> classOfPackagingContainingFile, String comp
 	}
 
 	public String toPath(Class<?> classOfPackagingContainingFile, String filename) {
-		return classOfPackagingContainingFile.getResource(filename).getPath();
+		String path = classOfPackagingContainingFile.getResource(filename).getPath();
+		if (path.endsWith(".java.template")) {
```

Comment:
This is just a file with`.template` extension that should be actually a `.java` file.

The issue here is that intellij somehow recognized our custom [`Context.java`](https://github.com/androidannotations/androidannotations/blob/7d53eb979de70096cf8033ddaf64b74cc8b01045/AndroidAnnotations/androidannotations-core/androidannotations/src/test/resources/org/androidannotations/generation/Context.java.template) as real class and thus caused the `automatically add @Override` feature of the save actions plugin to add `@Override` to three methods that [should not have it](https://github.com/androidannotations/androidannotations/pull/2176/commits/9f7e75d253d63312e5c4efe8c39558dc0bdee54f), as our `Context` class is not used for actuall compilation.

Reply:
Hmm, it do not really like we change code because of the IDE. Is this the only solution? Is this expected we will change those files soon?

Reply:
I can understand that it is not nice to change the code because of the IDE. Maybe we should open an issue with jetbrains that the given problem exists.

It is the only solution i found so far.... unfortunately we somehow need this class to test the behaviour in case the method we define in it is present on classpath. I wish we could get rid of these `*.java` in test resources at all. :/


No I do not think we can expect to change the files soon. It only exists for some tests that require framework classes to have certain methods that are not yet on API 16 (this should be our current build target IIRC).

Reply:
I would just turn off `automatically add @Override` feature. The IDE already does it anyway when you implement / override a method.

Reply:
okay, so i should turn of adding override annotation and revert the template stuff?

Reply:
Yes, please.

Reply:
updated.